Wrappers/Python/cil/framework/labels.py

```diff
@@ -0,0 +1,434 @@
+#  Copyright 2024 United Kingdom Research and Innovation
+#  Copyright 2024 The University of Manchester
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+# Authors:
+# CIL Developers, listed at: https://github.com/TomographicImaging/CIL/blob/master/NOTICE.txt
+
+from enum import Enum, EnumMeta
+
+class _LabelsBase(Enum):
+    """
+    Base class for labels enumeration.
+
+    Methods:
+    --------
+    - validate(label): Validates if the given label is a valid member of the enumeration.
+    - get_enum_member(label): Returns the enum member corresponding to the given label.
+    - get_enum_value(label): Returns the value of the enum member corresponding to the given label.
+    - __eq__(other): Checks if the enum value is equal to the given value.
+    - __contains__(item): Checks if the enum contains the given value.
+    """
+
+    @classmethod
+    def validate(cls, label):
+        """
+        Validates if the given label is a valid member of the specified class.
+
+        Parameters:
+        ----------
+        label : str, enum
+            The label to validate.
+
+        Raises:
+        -------
+        ValueError
+            If the label is not a valid member of the class.
+
+        Returns:
+        --------
+        bool
+            True if the label is a valid member of the class, False otherwise.
+        """
+
+        try:
+            for member in cls:
+                if member == label:
+                    return True
+        except AttributeError:
+            pass
+
+        raise ValueError(f"Expected one of {[e.value for e in cls]} \
+                         from {cls.__name__}, got {label}")
+
+
+    @classmethod
+    def get_enum_member(cls, label):
```

Comment:
not sure that this method is necessary as it's a one-liner to achieve same result anyway